azure_function/resource-graph-collector/__init__.py

```diff
@@ -44,19 +60,36 @@ def main(event: func.TimerRequest) -> None:
         logger.debug(f"Query results dump:\n---\n{result.as_dict()}\n---")
 
         # publish to loki
-        loki_logger = LokiPublisher(loki_endpoint=os.environ['LOKI_ENDPOINT'],
-                                    auth=(os.environ['LOKI_USERNAME'], os.environ['LOKI_PASSWORD']),
-                                    tags={'inventory_type': 'reference_architecture',
-                                        'graph_query_name': resource_graph_query.name})
-
-        fields_to_labels_string = os.environ.get('LOKI_LABEL_NAMES', None)
-        fields_to_labels = []
-        if fields_to_labels_string:
-            fields_to_labels = [e.strip() for e in fields_to_labels_string.split(',')]
-
-        if len(fields_to_labels) > MAX_LABELS:
-            raise ValueError(f"A maximum of {MAX_LABELS} is supported. You requested: {len(fields_to_labels)} labels, "
-                            f"namely: [{', '.join(fields_to_labels)}]")
-
-        for item in result.as_dict().get('data', []):
-            loki_logger.publish(item, fields_to_labels)
+        if is_enabled(os.environ.get("ENABLE_LOKI_PUBLISHER", "true")):
```

Comment:
I chose this approach having the backward compatibility in mind. So even if they inform the new environment variable the default value will be used, which I've deliberately chose "true" for Loki and "false" for Azure. WDYT?

Reply:
LGTM